pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -362,17 +199,166 @@ func (b *bootstrapper) WaitCleanup(ctx context.Context) error {
 	return managedresources.WaitUntilDeleted(timeoutCtx, b.client, b.namespace, b.name())
 }
 
+func (b *bootstrapper) deleteOldDWDManagedResource(ctx context.Context) error {
+	oldManagedResourceName := ManagedResourceDependencyWatchdogProbe
+	if b.values.Role == RoleWeeder {
+		oldManagedResourceName = ManagedResourceDependencyWatchdogEndpoint
+	}
+	if err := managedresources.DeleteForSeed(ctx, b.client, b.namespace, oldManagedResourceName); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("could not delete managed resource %s: %w", oldManagedResourceName, err)
+	}
+	if err := b.waitUntilManagedResourceDeleted(ctx, oldManagedResourceName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *bootstrapper) waitUntilManagedResourceDeleted(ctx context.Context, name string) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+	if err := managedresources.WaitUntilDeleted(timeoutCtx, b.client, b.namespace, name); err != nil {
+		return fmt.Errorf("error while waiting for managed resource %s to be deleted: %w", name, err)
+	}
+	return nil
+}
+
+func (b *bootstrapper) getConfigMap() (*corev1.ConfigMap, error) {
+	var (
+		config string
+		err    error
+	)
+
+	switch b.values.Role {
+	case RoleWeeder:
+		config, err = encodeConfig(&b.values.WeederConfig)
+		if err != nil {
+			return nil, err
+		}
+
+	case RoleProber:
+		config, err = encodeConfig(&b.values.ProberConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name() + "-config",
+			Namespace: b.namespace,
+			Labels:    map[string]string{v1beta1constants.LabelApp: b.name()},
+		},
+		Data: map[string]string{configFileName: config},
+	}
+	utilruntime.Must(kubernetesutils.MakeUnique(configMap))
+
+	return configMap, nil
+}
+
+func (b *bootstrapper) getServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name(),
+			Namespace: b.namespace,
+		},
+		AutomountServiceAccountToken: pointer.Bool(false),
+	}
+}
+
+func (b *bootstrapper) getClusterRoleAndClusterRoleBinding(serviceAccNamespace, serviceAccName string) (*rbacv1.ClusterRole, *rbacv1.ClusterRoleBinding) {
```

Comment:
Don't use artificial abbreviations not used anywhere else in this code base. Maybe you just want to pass `serviceAccount *corev1.ServiceAccount`?